src/Morphic-Widgets-Windows/SystemWindow.class.st

```diff
@@ -1277,10 +1277,12 @@ SystemWindow >> maximumExtent: aPoint [
 { #category : #api }
 SystemWindow >> menu [
 
-	| aMenu |
-	aMenu := MenuMorph new.
-	self windowMenuOn: aMenu.
-	^ aMenu
+	| aMenuBuilder |
+	aMenuBuilder := (PragmaMenuBuilder pragmaKeyword: self discoveredMenuPragmaKeyword model: self)
+		menuSpec;
+		yourself.
+				
+	^ aMenuBuilder menu
```

Comment:
I invoke @tesonep here for review, because he was chasing performance problems with the pragma collector, particularly in big images.

Reply:
I see... Then we need smarter caches system for pramgas.

Reply:
Yes, this change should be redone to not use pragmas.
Directly we can call the builder method

Reply:
I think the best solution is to use the builder and remove the usage of the window pragma.

Reply:
hmm, it follows the very same pattern as the world menu. Will we change the registration mechanism for both? (Commander or so...)

Reply:
ehhh.... guess what... we do not have the other working menu builder than the `PragmaMenuBuilder`

Reply:
Indeed based on Commander2 we wanted to build a new menu builder and remove the old one and esteban told us that this is not needed because people should use Spec2.
Well reality is in the middle. The pragmaMenuBuilder and friends suck quite a bit to my eyes and commander2 was designed (and the last chapter explains it) so that people can exactly build menu builder for their own framework.

Reply:
in this case, it is part of the Morphic-Core so we should take care of dependencies